src/Nethermind/Nethermind.Blockchain/OneTimeProcessor.cs

```diff
@@ -25,14 +25,16 @@ namespace Nethermind.Blockchain
     public class OneTimeChainProcessor : IBlockchainProcessor
     {
         private readonly IBlockchainProcessor _processor;
+        private readonly IBlockchainProcessor _mainChainProcessor;
         private readonly IReadOnlyDbProvider _readOnlyDbProvider;
 
         private object _lock = new object();
 
-        public OneTimeChainProcessor(IReadOnlyDbProvider readOnlyDbProvider, IBlockchainProcessor processor)
+        public OneTimeChainProcessor(IReadOnlyDbProvider readOnlyDbProvider, IBlockchainProcessor processor, IBlockchainProcessor mainChainProcessor = null)
         {
```

Comment:
can we pass some simple interface instead of a full additional processor? (an imterface with these two events only)

Reply:
Extracted interface